### ISSUE TITLE
Fake joins are not detected for call activities and expanded sub-processes.

### ISSUE THREAD

**Issue by timKraeuter** [bpmn-io/bpmnlint#125]
### Describe the Bug

Fake joins are not detected for call activities and expanded sub-processes. Example for call activities:

![fakeJoin_callActivity](https://github.com/bpmn-io/bpmnlint/assets/21026858/bcabfc5f-3d59-4c09-b28f-41edf220bd01)

The [fake-join rule](https://github.com/bpmn-io/bpmnlint/blob/main/rules/fake-join.js) only checks tasks and events.
The same applies to the [no-implicit-split rule](https://github.com/bpmn-io/bpmnlint/blob/main/rules/no-implicit-split.js). It only checks tasks and events.

**Possible fix:** Check all "bpmn:Activity" nodes instead of "bpmn:Task" in the rule since call activities and expanded sub-processes are activities.
I can draft a pull request with the fix (including new tests) if you like.

### Steps to Reproduce

Run:
``` bash
bpmn-lint fakeJoin_callActivity.txt
```
Or:
``` bash
bpmn-lint fakeJoin_Subprocess.txt
```
With the following BPMN-files:
[fakeJoin_callActivity.txt](https://github.com/bpmn-io/bpmnlint/files/13740605/fakeJoin_callActivity.txt)
[fakeJoin_Subprocess.txt](https://github.com/bpmn-io/bpmnlint/files/13740606/fakeJoin_Subprocess.txt)

### Expected Behavior

Fake joins should be detected for call activities and expanded sub-processes.

### Environment

- Host: npm CLI bpmn-lint
- OS: Windows 10
- Library version: 9.2

**Comment by nikku** [comment]
Great find. Happy to take a contribution that fixes this.

**Comment by timKraeuter** [comment]
Alright, done see #126.
